modules/calib3d/src/calibration.cpp

```diff
@@ -3855,6 +3855,91 @@ static void adjust3rdMatrix(InputArrayOfArrays _imgpt1_0,
     P3.at<double>(1,3) *= a;
 }
 
+static Mat homogeneousInverse(const Mat& T)
+{
+    CV_Assert( T.rows == 4 && T.cols == 4 );
+
+    Mat R = T(Rect(0, 0, 3, 3));
+    Mat t = T(Rect(3, 0, 1, 3));
+    Mat Rt = R.t();
+    Mat tinv = -Rt * t;
+    Mat Tinv = Mat::eye(4, 4, T.type());
+    Rt.copyTo(Tinv(Rect(0, 0, 3, 3)));
+    tinv.copyTo(Tinv(Rect(3, 0, 1, 3)));
```

Comment:
Consider using of `Matx44d` for well known mat sizes (including input/output parameters).